docs/how-to/test-dataset-definition.md

```diff
@@ -0,0 +1,186 @@
+
+## Example dataset definition
+
+First, we're creating a dataset definition with the following specifications:
+
+- The population includes everyone above the age of 18 on the `index_date` (31st March 2023).
+- The dataset has the following three columns:
+    - `age`: Patient's age on the `index_date`.
+    - `has_asthma_med`: Boolean value specifying whether a patient has received an asthma medication before the `index_date`.
+    - `latest_asthma_med_date`: Date of the most recent prescription of an asthma medication before the `index_date`.
+
+```ehrql
+from ehrql import create_dataset
+from ehrql.tables.core import patients, medications
+
+asthma_codes = ["39113311000001107", "39113611000001102"]
+
+dataset = create_dataset()
+
+index_date = "2023-03-31"
+
+dataset.age = patients.age_on(index_date)
+dataset.define_population(dataset.age > 18)
+
+latest_asthma_med = (
+    medications.where(medications.dmd_code.is_in(asthma_codes))
+    .where(medications.date <= index_date)
+    .sort_by(medications.date)
+    .last_for_patient()
+)
+
+dataset.has_asthma_med = latest_asthma_med.exists_for_patient()
+dataset.latest_asthma_med_date = latest_asthma_med.date
+```
+
+## Specifying data and expectations
+
+Next, you need to provide (1) data for test patients and (2) define the expected outcomes for these patients after applying your ehrQL queries.
+Test data and expectations are defined in a nested dictionary called `patient_data`.
+The key in the outermost dictionary specifies the patient id.
+
+### Data for test patients
+
+In the example below we have created one test patient with the patient id `1` and specified more data for this patient in two lists called `patient` and `medications`.
+The names of the inner lists match the names of the tables you are using in your dataset definition.
+To explore how these tables are structured you can look at the column names in the [core schema](../../reference/schemas/core) documentation.
+You can also generate CSV files of these tables and add them to your project using the [dump-example-data](../../reference/cli/#dump-example-data) or [create-dummy-tables](../../reference/cli/#create-dummy-tables) commands.
+
+The `patients` list contains one dictionary with two keys, one key for each column (`date_of_birth` and `sex`) that we want to create.
+Note that you don't have to specify a value for each column in the underlying table.
+For example we did not specify `date_of_death` in the dictionary inside the `patients` list so the column will be missing with the value `None`.
+
+The `medications` list contains two dictionaries (one for each row we add to the table) with two keys (one for each column: `date` and `dmd_code`).
+Note that you have to specify a list for each table you import from `ehrql.tables.core` in your dataset definition, but this could also be an empty list.
```

Comment:
Not quite true: you have to specify a list for each table you use in your dataset definition.  You could import a table and then not use it.

Reply:
thanks, that makes even simpler to describe!